src/core/resources/transactions/transaction.ts

```diff
@@ -83,133 +94,93 @@ export const fetchTransaction = async ({
   const value = formatUnits(transaction.value, decimals);
 
-  const parsedTransaction = transaction.blockNumber
-    ? ({
-        status: 'confirmed',
-        blockNumber: transaction.blockNumber || 0,
-        minedAt: block?.timestamp || 0,
-        confirmations: transaction.confirmations,
-        gasUsed: transaction.gasLimit.toString(),
-        hash: transaction.hash as Hash,
-        nonce: transaction.nonce,
-        chainId: transaction.chainId,
-        from: transaction.from as Address,
-        to: transaction.to as Address,
-        data: transaction.data,
-        value,
-        type: 'send',
-        title: i18n.t('transactions.send.confirmed'),
-        baseFee: block?.baseFeePerGas?.toString(),
-        maxFeePerGas: transaction.maxFeePerGas?.toString(),
-        maxPriorityFeePerGas: transaction.maxPriorityFeePerGas?.toString(),
-        gasPrice: transaction.gasPrice?.toString(),
-      } satisfies MinedTransaction)
-    : ({
-        status: 'pending',
-        hash: transaction.hash as Hash,
-        nonce: transaction.nonce,
-        chainId: transaction.chainId,
-        from: transaction.from as Address,
-        to: transaction.to as Address,
-        data: transaction.data,
-        value,
-        type: 'send',
-        title: i18n.t('transactions.send.pending'),
-      } satisfies PendingTransaction);
-  return parsedTransaction;
+  const direction =
+    transaction.from === account ? ('out' as const) : ('in' as const);
+
+  const baseTransaction = {
+    hash: transaction.hash as Hash,
+    nonce: transaction.nonce,
+    chainId: transaction.chainId,
+    from: transaction.from as Address,
+    to: transaction.to as Address,
+    data: transaction.data,
+    value,
+    direction,
+
+    feeType: !transaction.maxFeePerGas ? 'legacy' : 'eip-1559',
+    gasLimit: transaction.gasLimit.toString(),
+    maxFeePerGas: transaction.maxFeePerGas?.toString(),
+    maxPriorityFeePerGas: transaction.maxPriorityFeePerGas?.toString(),
+    gasPrice: transaction.gasPrice?.toString(),
+  } as const;
+
+  if (transaction.blockNumber !== undefined) {
+    const [receipt, block, type] = await Promise.all([
+      provider.getTransactionReceipt(transaction.hash),
+      provider.getBlock(transaction.blockNumber),
+      guessTransactionType(provider, transaction),
+    ]);
+    const status = receipt.status === 1 ? 'confirmed' : 'failed';
+
+    return {
+      ...baseTransaction,
+      status,
+      type,
+      title: i18n.t(`transactions.${type}.${status}`),
+
+      blockNumber: transaction.blockNumber,
+      minedAt: transaction.timestamp!,
+      confirmations: transaction.confirmations,
+
+      gasUsed: receipt.gasUsed.toString(),
+
+      feeType: !block.baseFeePerGas ? 'legacy' : 'eip-1559',
+      fee: receipt.cumulativeGasUsed.mul(receipt.effectiveGasPrice).toString(),
+      baseFee: block.baseFeePerGas?.toString(),
+    };
+  }
+
+  const type = await guessTransactionType(provider, transaction);
```

Comment:
why await?

Reply:
Im fetching data from the provider to guess the type
like `transaction.to` has code (`provider.getCode`) means it's probably a transaction interacting with a contract

Reply:
hmm idk whats the expected behavior on smart wallets